spring-cloud-loadbalancer/src/main/java/org/springframework/cloud/loadbalancer/config/LoadBalancerStatsAutoConfiguration.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2012-2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.loadbalancer.config;
+
+import io.micrometer.core.instrument.MeterRegistry;
+
+import org.springframework.boot.autoconfigure.condition.ConditionalOnBean;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnClass;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.cloud.loadbalancer.stats.MicrometerStatsLoadBalancerLifecycle;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+/**
+ * Autoconfiguration that provides a {@link MicrometerStatsLoadBalancerLifecycle} bean.
+ *
+ * @author Olga Maciaszek-Sharma
+ * @since 3.0.0
+ */
+@Configuration(proxyBeanMethods = false)
+@ConditionalOnClass(MeterRegistry.class)
+@ConditionalOnProperty(value = "spring.cloud.loadbalancer.stats.micrometer.enabled", havingValue = "true")
+public class LoadBalancerStatsAutoConfiguration {
```

Comment:
Should this be auto configuration (global) or per loadbalancer?

Reply:
I think it should be global as the actual metrics bean can support `N` `serviceId`s, but we can discuss it. Do you think it should be per load-balancer?

Reply:
If we added it per-loadbalancer, though, we would have one bean per `serviceId` and, therefore, create separate meters per `seviceId` and also separate maps. Let me know what you think.

Reply:
I think it's fine for now. The real issue about memory we can worry about later.

Reply:
Ok. Will merge it then.